### DISCUSSION TITLE
Is it possible to decode a message over time?

### DISCUSSION THREAD

**Discussion by viniciusth** [tokio-rs/prost#790] [Q&A]
As the title says, I want to know if it is possible to keep a Decoder in memory and keep feeding it bytes of a message as I receive them, but the only way I found to decode currently is to have the entire buffer at once and decode it.
It would be something like this:

```rust
let mut decoder = MessageDecoder::<MyType>::new(..);
while still_has_bytes {
  ... // do some stuff while waiting for bytes from another thread
  decoder.write(bytes); // got the bytes and wrote to the decoder
}
let message = decoder.finish();
```

**Comment by LucioFranco** [comment]
You can keep trying to decode, if it fails you can wait for more bytes.
